### ISSUE TITLE
[BUG] write_with_options fails with radix 16 and power-of-two enabled

### ISSUE THREAD

**Issue by Atmelfan** [Alexhuszagh/rust-lexical#71]
## Description

This code ($radix = 16):

```rust
let mut buf = [b'0'; <$typ>::FORMATTED_SIZE];
const FORMAT: u128 = NumberFormatBuilder::from_radix($radix);
let options = lexical_core::WriteIntegerOptions::new();
let slc = lexical_core::write_with_options::<_, FORMAT>(self.0, &mut buf, &options);
```

Fails with `thread 'integer_i16::test_hex' panicked at 'assertion failed: NumberFormat::<{ FORMAT }>::MANTISSA_RADIX <= 10', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/lexical-parse-integer-0.8.0/src/algorithm.rs:288:5`.

## Prerequisites

Here are a few things you should provide to help me understand the issue:

- Rust version: `rustc 1.58.0-nightly`
- lexical version: lexical-core 0.8.0
- lexical compilation features used: write-float, write-integer, parse-float, parse-integer, power-of-two

## Test case

Example test case:

```
use lexical_core;

fn main() {
        let mut buf = [b'0'; i16::FORMATTED_SIZE];
        const FORMAT: u128 = NumberFormatBuilder::from_radix(16);
        let options = lexical_core::WriteIntegerOptions::new();
        let slc = lexical_core::write_with_options::<_, FORMAT>(1234, &mut buf, &options);
}
```

## Additional Context

Works with `radix` feature enabled but it would be nice if hexadecimal worked in power-of-two or mentioned as not supported in the docs.

**Comment by Alexhuszagh** [comment]
Interesting, this should be covered by my test cases. I'll check the internals on this shortly.

**Comment by Alexhuszagh** [comment]
Ok that's interesting, since it's clearly using an unexpected code path. The relevant code is here:

https://github.com/Alexhuszagh/rust-lexical/blob/dbc42cd12205da1c8821e67fb9f9c21cd31d87b8/lexical-parse-integer/src/algorithm.rs#L288

Which `try_parse_8digits` should only be called if the `radix <= 10` (since we can't use these optimizations otherwise).

**Comment by Alexhuszagh** [comment]
Ah, it seems the issue is here:

https://github.com/Alexhuszagh/rust-lexical/blob/dbc42cd12205da1c8821e67fb9f9c21cd31d87b8/lexical-parse-integer/src/algorithm.rs#L24-L29

This should be changed to use `power-of-two`, instead of `radix`, as the feature. I'll also update the unittests to ensure this is covered.

**Comment by Alexhuszagh** [comment]
Fixed as of 64fe7aaa18987c34a8ff506368a9eaef3522493f.

**Comment by Alexhuszagh** [comment]
Fixed in the released [v0.8.3](https://crates.io/crates/lexical-core/0.8.3).
